src/util/usertools_extra.c

```diff
@@ -0,0 +1,53 @@
+/*
+    SSSD
+
+    usertools_extra.c
+
+    Authors:
+        Sumit Bose <sbose@redhat.com>
+        Iker Pedrosa <ipedrosa@redhat.com>
+
+    Copyright (C) 2021 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <pwd.h>
+#include <errno.h>
+#include <talloc.h>
+#include <grp.h>
+
+#include "db/sysdb.h"
+#include "confdb/confdb.h"
+#include "util/strtonum.h"
+#include "util/util.h"
+#include "util/safe-format-string.h"
+#include "responder/common/responder.h"
+
+errno_t sss_load_nss_pw_symbols(struct sss_nss_ops *ops)
```

Comment:
This makes cwrap testing possible. Without it, the testing infrastructure would load the system's getpw*_r() functions instead of the mocked ones, which use `sssd/src/tests/cwrap/passwd` as the user database.

With that said, I'm open to other ways of solving this problem.

Reply:
Let's keep this conversation open awhile (until other comments are resolved).

Reply:
Ok, I wasn't able to come up with anything more elegant.
Messing with LD_PRELOAD here feels like shooting own leg....

Reply:
But maybe `nss_dl_load_extra` could be a better name for this file.

Reply:
Yes, it makes more sense to call it `nss_dl_load_extra.c`